pkg/apis/core/validation/cloudprofile.go

```diff
@@ -410,3 +412,101 @@ func validateRegions(regions []core.Region, fldPath *field.Path) field.ErrorList
 
 	return allErrs
 }
+
+func validateBastion(spec *core.CloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	var (
+		allErrs     field.ErrorList
+		machineArch *string
+	)
+
+	if spec.Bastion == nil {
+		return allErrs
+	}
+
+	if spec.Bastion.MachineType != nil {
+		var validationErrors field.ErrorList
+		machineArch, validationErrors = validateBastionMachineType(spec.Bastion.MachineType, spec.MachineTypes, fldPath.Child("machineType"))
+		allErrs = append(allErrs, validationErrors...)
+	}
+
+	if spec.Bastion.MachineImage != nil {
+		allErrs = append(allErrs, validateBastionImage(spec.Bastion.MachineImage, spec.MachineImages, machineArch, fldPath.Child("machineImage"))...)
+	}
```

Comment:
Is it valid to have a bastion in the cloud profile without a machine type or machine image? The validation code allows this, but I do not understand why.

Reply:
Good catch!

Reply:
Is it is much different than the nil case ? or do we consider the empty struct a misconfiguration?

Reply:
In the end the bastion controller would not care and handle it the same way as if the bastion section is not set.
But isn't it still a configuration that should not be allowed because you either set it to something meaningful or not?